src/java/com/sarality/app/loader/DataLoaderTask.java

```diff
@@ -21,7 +21,10 @@ public DataLoaderTask(Context context, DataSource<T> dataSource) {
 
   @Override
   public T loadInBackground() {
-    dataSource.loadData();
-    return dataSource.getData();
+    if (dataSource != null) {
```

Comment:
If a Fragment or Initializer is initialzed without requiring a datasource, the getDataSource methods can simply return null. This way when loadInBackground is called we don't fail with a NPE but instead call the consumer with null.